src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareLanguageServerClient.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
+using Nerdbank.Streams;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
+{
+    [ContentType("C#_LSP")]
+    [ContentType("VB_LSP")]
+    [Export(typeof(ILanguageClient))]
+    internal class LiveShareLanguageServerClient : ILanguageClient
+    {
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly IThreadingContext _threadingContext;
+        private readonly LanguageServerProtocol _languageServerProtocol;
+        private readonly Workspace _workspace;
+
+        /// <summary>
+        /// Gets the name of the language client (displayed to the user).
+        /// </summary>
+        public string Name => ServicesVSResources.LiveShare_CSharp_Visual_Basic_Language_Server_Client;
+
+        /// <summary>
+        /// Gets the configuration section names for the language client. This may be null if the language client
+        /// does not provide settings.
+        /// </summary>
+        public IEnumerable<string>? ConfigurationSections { get; } = null;
+
+        /// <summary>
+        /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
+        /// This may be null if the client does not need custom initialization options.
+        /// </summary>
+        public object? InitializationOptions { get; } = null;
+
+        /// <summary>
+        /// Gets the list of file names to watch for changes.  Changes will be sent to the server via 'workspace/didChangeWatchedFiles'
+        /// message.  The files to watch must be under the current active workspace.  The file names can be specified as a relative
+        /// paths to the exact file, or as glob patterns following the standard in .gitignore see https://www.kernel.org/pub/software/scm/git/docs/gitignore.html files.
+        /// </summary>
+        public IEnumerable<string>? FilesToWatch { get; } = null;
```

Comment:
Should the comment indicate that we don't actually use this (and similar for all the other ones we aren't using)?

Reply:
I'd also love comments as to _why_ we don't implement them....